src/components/UserSelector.tsx

```diff
@@ -1,16 +1,47 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { User } from '../types/User';
+import { getUsers } from './api/users';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onSelectedUser: (val: User | null) => void;
+  selectedUser: User | null;
+  onSelectedPost: (val: Post | null) => void;
+};
+
+export const UserSelector: React.FC<Props> = ({
+  selectedUser,
+  onSelectedUser,
+  onSelectedPost,
+}) => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  useEffect(() => {
+    getUsers().then(setUsers);
+  }, []);
+
+  const handleSelectUser = (user: User) => {
+    onSelectedPost(null);
+    onSelectedUser(user);
+  };
```

Comment:
The function `handleSelectUser` is not following the naming conventions mentioned in the checklist. It's better to name event handlers as 'handle' + 'what it handles', for example, in this case, it could be `handleUserSelect`.